graftm/rerooter.py

```diff
@@ -61,50 +60,24 @@ def reroot(self, tree):
         
         Parameters
         ----------
-        tree: TreeNode object
-            TreeNode object (tree opened with skbio)
+        tree: dendropy.Tree
+            Tree to be rerooted
             
         Raises
         ------
         BinaryTreeError : If the input tree is already a binary tree
 
         '''
 
-        children = tree.children
-        no_found_root = True
+        children = {x.length:x for x in tree.seed_node.child_edges()}
+        tree.is_rooted=True
         
         if len(children) > 2:
-            
-            node_combinations = list(itertools\
-                                   .combinations([x for x in children], 
-                                                 2))
-
-            length_combination = list(itertools\
-                                   .combinations([x.length for x in children], 
-                                                 2))
-            
-            sums = [x[0] + x[1] for x in length_combination]
-            
-            while no_found_root:
-                
-                nodes = [node_combinations[sums.index(x)]
-                         for x in heapq.nlargest(2, sums)]
-                most_distant_node = [node for node in nodes[0] 
-                                     if node in nodes[1]]
-                if len(most_distant_node) > 0:
-                    most_distant_node = most_distant_node[0]
-                    no_found_root = False
-                
-            other_nodes = [node for node in children 
-                           if node != most_distant_node]
-            
-            dec_len = most_distant_node.length/2
-            
-            most_distant_node.length = dec_len
-            link_root = TreeNode(children=other_nodes, length = dec_len)
-            tree = TreeNode(children=[link_root, most_distant_node], 
-                            name = "root",
-                            length=0)
+            longest_edge_length = max(children.keys())
```

Comment:
I suppose this achieves the aim of choosing an edge arbitrarily if there are 2 equally maximum edge lengths, so ok.